libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -74,4 +76,36 @@ export class StripeManager {
       customer.tax?.automatic_tax === 'not_collecting'
     );
   }
+
+  /**
+   * If customer taxId does not match incoming taxId, update customer record with incoming taxId.
+   *
+   * @param customerId
+   * @param taxId
+   * @returns
+   */
+  async addTaxIdToCustomer(customerId: string, taxId: string) {
+    // fetch customer by customerId
+    const customer = await this.client.fetchCustomer(customerId);
+
+    // check that customer exists
+    if (!customer) {
+      throw new Error('customer does not exist');
+    }
+
+    // check that customer hasn't been deleted
+    if (customer.deleted) throw new CustomerDeletedError();
+
+    // check that customer does not already have taxId
+    if (!customer?.tax_ids && !customer.tax_ids?.object.includes(taxId)) {
```

Comment:
Does this need the `?` in the first `customer?.tax_ids`? IIUC `customer` should always be truthy since we already yolo out on L92 above if the customer does not exist.

Reply:
I was wondering about this. I get an error though about `tax_ids` and `object` possibly being undefined when I don't include it. I am all for updating the way it's written because it looks odd. Maybe there is a better way to do the check? Open to any and all ideas! :mozheart:

Reply:
This `if` statement can be replaced with `customer.tax_ids?.object.includes(taxId)`

Reply:
Thank you @julianpoy ! As per our Slack discussion, I have updated to split into `setCustomerTaxId` and `getCustomerTaxId` respectively, updated tests, and added additional information to this PR in the description above. More than happy to make further optimizations! Please let me know! 🙏 🥂